lib/raap/cli.rb

```diff
@@ -274,10 +274,11 @@ def property(receiver_type:, type_params_decl:, type_args:, method_type:, method
       end
 
       # type_args delegate to self_type
-      type_params_decl.each_with_index do |param, i|
-        if rtype.instance_of?(::RBS::Types::ClassInstance)
-          rtype.args[i] = type_args[i] || param.upper_bound || ::RBS::Types::Bases::Any.new(location: nil)
+      if rtype.instance_of?(::RBS::Types::ClassInstance)
+        args = type_params_decl.map.with_index do |param, i|
+          type_args[i] || param.upper_bound || ::RBS::Types::Bases::Any.new(location: nil)
         end
+        receiver_type.type = rtype = ::RBS::Types::ClassInstance.new(name: rtype.name, args: args, location: rtype.location)
```

Comment:
This change would eliminate the need to change the accessibility of `RaaP::Type#type`.

```suggestion
        rtype = ::RBS::Types::ClassInstance.new(name: rtype.name, args: args, location: rtype.location)
        receiver_type = Type.new(rtype)
```

Reply:
Thanks for your review! I've modified it to avoid using `attr_accessor` for `Raap::Type` 
https://github.com/ksss/raap/pull/25/commits/2390a0ee421675cd01879c32778aaee53854fb02